rllab/plotter/plotter.py

```diff
@@ -60,6 +60,8 @@ def init_worker():
 
 
 def init_plot(env, policy):
+    if queue is None:
+        init_worker()
```

Comment:
Whose responsibility is it to call init_worker()?

1. This code suggests that init_plot() does a silent, idempotent initialization of the process-global state, and init_worker() is only ever called by init_plot() internally.
1. Elsewhere, you call init_worker() from the algorithm (in BatchPolopt)
1. Other parts of the framework (e.g. run_experiment_lite) call init_worker() as part of global process setup.

These are three different strategies for initializing the worker. All are defensible. Right now, the framework uses (3). It is fine to change the initialization API codebase-wide if it makes sense, but it the convention needs to be consistent.

Reply:
Actually, while verifying some algorithms , I observed that they were not initializing init_worker. an exception was raised. So to protect from those conditions , I made sure that init_worker is called at least once. 
Do I need to add this initialization in all run_experiment_lite or launcher code? I didn't want to modify the existing code. Hence kept the code as it is with little modification.

Reply:
Please let me know what needs to be done here. I have made the changes which you have suggested. After completing this I will again raise a request.

Reply:
There is no bug in the algorithms perse.

The API contract for `plotter` is that the user scripts have to call `plotter.init_worker()` before calling `algo.train()`. This is usually done for the user inside the wrapper `scripts/run_experiment_lite.py`. Currently, if the user doesn't use `run_experiment_lite`, they have to call `plotter.init_worker()` manually.

The examples you were using probably had plotting turned off (`plot=False`), and so didn't include `plotter.init_worker()` before `algo.train()`. Your are correct--the user must always call `plotter.init_worker()` in either `run_experiment_lite()` or the launcher before `algo.train()` IF they specify `algo.train(..., plot=True, ...)`.

We might want to change the API contract, but this is the contract right now. It is not a bug. I discuss changing the API contract below.